src/components/ChatLog.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import './ChatLog.css';
+import ChatEntry from './ChatEntry';
+import './ChatLog.css'
+import PropTypes from 'prop-types';
+
+const ChatLog = (props) => {
+    const chatLog = 'chat-log';
+    let showLike;
+    console.log(props.entries);
+    const chatComponents = props.entries.map((chat, index)=> {
+        const entriesData = [...props.entries];
+        const totalLike = [...entriesData].reduce((a,b) => a + (b.liked? 1: 0), 0)
+        showLike = totalLike === 0? '🤍' : totalLike + ' ❤️s'; 
```

Comment:
The logic for counting the liked messages would be better placed in App. The mockups showed that the intended location for displying the liked count was as part of the page header rather than being an instrinsic part of the conversation. Yes, we can add heading tags her in the ChatLog, but the main responsibility of the ChatLog is to generate the indivdual ChatEntry components and wireup their events properly.

Note that by having this here in the `map` callback, that means all the messages will be iterated over for each message being converted into a component, making this a O(n^2) operation. We could at the least move this logic out of the `map` callback and into the body of the component, but it would still be better located in App.